src/chains/ethereum/ethereum/tests/api/eth/eth.test.ts

```diff
@@ -543,34 +543,67 @@ describe("api", () => {
-      await provider.send("eth_subscribe", ["newHeads"]);
-      const txJson = {
-        type: "0x2",
-        chainId: "0x539",
-        nonce: "0x0",
-        from: accounts[0],
-        to: accounts[1],
-        value: "0x1",
-        maxPriorityFeePerGas: "0xf",
-        maxFeePerGas: "0xfffffffff",
-        gas: "0x15f90",
-        input: "0x01",
-        accessList: []
-      } as any;
-      const hash = await provider.send("eth_sendTransaction", [txJson]);
-      const _message = await provider.once("message");
-      // we want these values set for when we check against the return data,
-      // but they shouldn't be used in eth_sendTransaction, so we'll set them now
-      txJson.transactionIndex = "0x0";
-      txJson.gasPrice = "0x342770cf";
-
-      const tx = await provider.send("eth_getTransactionByHash", [hash]);
+    describe("eth_getTransactionByHash", () => {
+      it("returns a transaction matching the sent transaction", async () => {
+        await provider.send("eth_subscribe", ["newHeads"]);
+        const txJson = {
+          type: "0x2",
+          chainId: "0x539",
+          nonce: "0x0",
+          from: accounts[0],
+          to: accounts[1],
+          value: "0x1",
+          maxPriorityFeePerGas: "0xf",
+          maxFeePerGas: "0xfffffffff",
+          gas: "0x15f90",
+          input: "0x01",
+          accessList: []
+        } as any;
+        const hash = await provider.send("eth_sendTransaction", [txJson]);
+        const _message = await provider.once("message");
+        // we want these values set for when we check against the return data,
+        // but they shouldn't be used in eth_sendTransaction, so we'll set them now
+        txJson.transactionIndex = "0x0";
+        txJson.gasPrice = "0x342770cf";
+
+        const tx = await provider.send("eth_getTransactionByHash", [hash]);
+
+        // loop over all of the data we set to verify it matches
+        for (const [key, value] of Object.entries(txJson)) {
+          assert.deepStrictEqual(value, tx[key]);
+        }
+      });
 
-      // loop over all of the data we set to verify it matches
-      for (const [key, value] of Object.entries(txJson)) {
-        assert.deepStrictEqual(value, tx[key]);
-      }
+      it("has a `gasPrice` that matches the corresponding receipt's `effectiveGasPrice` after the transaction is mined", async () => {
+        await provider.send("eth_subscribe", ["newHeads"]);
+        await provider.send("miner_stop", []);
+        const txJson = {
+          type: "0x2",
+          from: accounts[0],
+          to: accounts[1],
+          maxPriorityFeePerGas: "0x77359400",
+          maxFeePerGas: "0x6FC23AC00"
+        } as any;
+        // we previously had a bug where the `gasPrice` field returned from
+        // `eth_getTransactionByHash` was incorrect when multiple transactions
+        // were sent, because we weren't saving the updated `effectiveGasPrice`.
+        // this tests against that. see:
```

Comment:
I think something like is more accurate according to the actual fix:

```suggestion
        // were sent, because we weren't assigning the correct `effectiveGasPrice`
        // when adding the transaction to the pool. see:
```